### PR TITLE
[RemoteUtil] Fix output from Windows SSH commands

### PR DESCRIPTION
**Pull request by ljusten**
Adds an ArchType argument to many RemoteUtil methods, which is used to replace -tt (forced pseudo-TTY allocation) by -T (no pseudo-TTY allocation). The -tt option adds tons of ANSI escape sequences to the output and makes it unparsable, even after removing the sequences, as some sequences like "delete the last X characters" are not honoured.

### CODE REVIEW

File: common/remote_util.h [hunk 1]
```diff
@@ -92,15 +93,18 @@ class RemoteUtil {
 
   // Runs |remote_command| on the remote device. The command must be properly
   // escaped. |name| is the name of the command displayed in the logs.
-  absl::Status Run(std::string remote_command, std::string name);
+  absl::Status Run(std::string remote_command, std::string name,
+                   ArchType arch_type);
 
   // Same as Run(), but captures both stdout and stderr.
   // If |std_out| or |std_err| are nullptr, the output is not captured.
   absl::Status RunWithCapture(std::string remote_command, std::string name,
```
Comment (PatriosTheGreat) [inline common/remote_util.h:101]: It's a bit non-obvious which arch type need to be passed here (the current machine one or the target machine). Can you please add a description here? (also maybe name a parameter corresponding local_arch_type)

File: common/remote_util_test.cc [hunk 2]
```diff
@@ -59,7 +59,8 @@ class RemoteUtilTest : public ::testing::Test {
 };
 
 TEST_F(RemoteUtilTest, BuildProcessStartInfoForSsh) {
-  ProcessStartInfo si = util_.BuildProcessStartInfoForSsh(kCommand);
+  ProcessStartInfo si =
+      util_.BuildProcessStartInfoForSsh(kCommand, ArchType::kLinux_x86_64);
   ExpectContains(si.command, {"ssh", kUserHostArg, kCommand});
```
Comment (PatriosTheGreat) [inline common/remote_util_test.cc:64]: Should we also expect -t here?
Also maybe it make sense to add a test case for kWindows_x86_64

**Review by PatriosTheGreat — CHANGES_REQUESTED** [review]

File: common/remote_util_test.cc (hunk 2 above)
Comment (ljusten) [reply common/remote_util_test.cc:64]: Done.

**Review by PatriosTheGreat — APPROVED** [review]
LGTM, thanks!

**Pull request merged**